### PR TITLE
Bug 1229642 - new-build generates ImageStreams along with BuildConfig when needed

### PR DESCRIPTION
**Pull request by soltysh**
This will fix https://bugzilla.redhat.com/show_bug.cgi?id=1229642

@smarterclayton ptal

### CODE REVIEW

**Comment by smarterclayton** [comment]
@csrwng or @kargakis review please

**Comment by kargakis** [comment]
@soltysh please update the comments of `Accept` and `NewAcceptBuildConfigs` to reflect their new functionality. Also don't we have those strings defined somewhere as constants?

**Comment by csrwng** [comment]
@soltysh - so this will output image streams for things that are not necessarily build input/outputs... is that desired? 
```
oc new-build https://github.com/csrwng/simple-ruby --docker-image=openshift/ruby-20-centos7 --docker-image=openshift/mongodb-24-centos7 -o yaml
```

**Comment by soltysh** [comment]
@csrwng after doing some tests, I've decided to move the filtering entirely to `newapp.go`, see `filterImageStreams` method, which is called after the `AppResult` is created to filter unnecessary images. ptal
@kargakis updated comments per your comment. ptal

**Comment by soltysh** [comment]
@csrwng your comments addressed, ptal.

**Comment by csrwng** [comment]
LGTM

**Comment by soltysh** [comment]
@smarterclayton can I get your approval for this?

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -476,8 +476,63 @@ func (c *AppConfig) RunBuilds(out io.Writer) (*AppResult, error) {
...
+			}
+		} else {
+			items = append(items, item)
+		}
+	}
+	result.List.Items = items
+	return result
+}
+
+func makeImageStreamKey(ref *kapi.ObjectReference) string {
```
Comment (smarterclayton) [inline pkg/generate/app/cmd/newapp.go]: We have imageapi.SplitImageStreamTag and JoinImageStreamTag for specifically this reason.  We have NamespacedName as well, use those three.

**Comment by smarterclayton** [comment]
Also, needs a test case.

**Comment by soltysh** [comment]
@smarterclayton updated current test case to check generated IS and changed how we create the keys. I think now it's ready to go. This may conflict with #3057 if it lands sooner.

[test] in the mean time.

**Comment by soltysh** [comment]
Ups, small test case failure, should be ok now.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**